plugins/outputs/elasticsearch/elasticsearch.go

```diff
@@ -28,26 +29,27 @@ import (
 var sampleConfig string
 
 type Elasticsearch struct {
-	AuthBearerToken     config.Secret   `toml:"auth_bearer_token"`
-	DefaultPipeline     string          `toml:"default_pipeline"`
-	DefaultTagValue     string          `toml:"default_tag_value"`
-	EnableGzip          bool            `toml:"enable_gzip"`
-	EnableSniffer       bool            `toml:"enable_sniffer"`
-	FloatHandling       string          `toml:"float_handling"`
-	FloatReplacement    float64         `toml:"float_replacement_value"`
-	ForceDocumentID     bool            `toml:"force_document_id"`
-	HealthCheckInterval config.Duration `toml:"health_check_interval"`
-	HealthCheckTimeout  config.Duration `toml:"health_check_timeout"`
-	IndexName           string          `toml:"index_name"`
-	ManageTemplate      bool            `toml:"manage_template"`
-	OverwriteTemplate   bool            `toml:"overwrite_template"`
-	Username            config.Secret   `toml:"username"`
-	Password            config.Secret   `toml:"password"`
-	TemplateName        string          `toml:"template_name"`
-	Timeout             config.Duration `toml:"timeout"`
-	URLs                []string        `toml:"urls"`
-	UsePipeline         string          `toml:"use_pipeline"`
-	Log                 telegraf.Logger `toml:"-"`
+	AuthBearerToken     config.Secret          `toml:"auth_bearer_token"`
+	DefaultPipeline     string                 `toml:"default_pipeline"`
+	DefaultTagValue     string                 `toml:"default_tag_value"`
+	EnableGzip          bool                   `toml:"enable_gzip"`
+	EnableSniffer       bool                   `toml:"enable_sniffer"`
+	FloatHandling       string                 `toml:"float_handling"`
+	FloatReplacement    float64                `toml:"float_replacement_value"`
+	ForceDocumentID     bool                   `toml:"force_document_id"`
+	HealthCheckInterval config.Duration        `toml:"health_check_interval"`
+	HealthCheckTimeout  config.Duration        `toml:"health_check_timeout"`
+	IndexName           string                 `toml:"index_name"`
+	IndexTemplate       map[string]interface{} `toml:"template_index_settings"`
```

Comment:
Hmmm why is this a map instead of a simple string?

Reply:
I figured this was easier for the config file representation, though I'm happy to change it if you think string makes more sense

Reply:
OK, I'm fine with your way.